### ISSUE TITLE
Categories confusion

### ISSUE THREAD

**Issue by peteruithoven** [Chadsr/NordVPN-NetworkManager#37]
As a fellow resident of the Netherlands I was trying to connect to one of the nl servers, but I ran into some issues:

### When I setup my settings with just nl
```
[Countries]
# simply write country codes separated by spaces e.g. country-blacklist = gb us
country-blacklist = 

# same as above. if this is non-empty, the blacklist is ignored
country-whitelist = nl

[Categories]
p2p = false
onion-over-vpn = false
anti-ddos = false
standard-vpn-servers = true
dedicated-ip-servers = false
double-vpn = false

[Protocols]
tcp = true
udp = false

[Benchmarking]
ping-attempts = 5
```

It has troubles getting the server list from NordVPN:
```
$ sudo nordnm -uska nl normal tcp
     _   _               _ _   _ ___  ___
    | \ | |             | | \ | ||  \/  |
    |  \| | ___  _ __ __| |  \| || .  . |
    | . ` |/ _ \| '__/ _` | . ` || |\/| |
    | |\  | (_) | | | (_| | |\  || |  | |
    \_| \_/\___/|_|  \__,_\_| \_/\_|  |_/   v0.0a13 (Latest)

[INFO] [nordnm.nordnm:160]: Downloading latest NordVPN OpenVPN configuration files to '/home/peteruithoven/.nordnm/configs/'.
[INFO] [nordnm.nordnm:307]: Checking for new connections to import...
[ERROR] [nordnm.nordnm:372]: Could not fetch the server list from NordVPN. Check your Internet connectivity.
```

### When I setup my settings with just us
```
[Countries]
# simply write country codes separated by spaces e.g. country-blacklist = gb us
country-blacklist = 

# same as above. if this is non-empty, the blacklist is ignored
country-whitelist = us

[Categories]
p2p = false
standard-vpn-servers = true
double-vpn = false
onion-over-vpn = false
anti-ddos = false
dedicated-ip-servers = false

[Protocols]
tcp = true
udp = false

[Benchmarking]
ping-attempts = 5
```
It does work:
```
$ sudo nordnm -uska us normal tcp
     _   _               _ _   _ ___  ___
    | \ | |             | | \ | ||  \/  |
    |  \| | ___  _ __ __| |  \| || .  . |
    | . ` |/ _ \| '__/ _` | . ` || |\/| |
    | |\  | (_) | | | (_| | |\  || |  | |
    \_| \_/\___/|_|  \__,_\_| \_/\_|  |_/   v0.0a13 (Latest)

[INFO] [nordnm.nordnm:160]: Downloading latest NordVPN OpenVPN configuration files to '/home/peteruithoven/.nordnm/configs/'.
[INFO] [nordnm.nordnm:307]: Checking for new connections to import...
[WARNING] [nordnm.nordnm:327]: Active VPNs and/or kill-switch disabled for accurate benchmarking. Your connection is not secure until these are re-enabled.
[INFO] [nordnm.nordnm:329]: Benchmarking servers...
[INFO] [nordnm.nordnm:337]: Benchmarking complete. Took 14.00 seconds.
[INFO] [nordnm.nordnm:201]: Auto-Connect file removed.
[INFO] [nordnm.nordnm:209]: Removing all active connections...
[INFO] [nordnm.nordnm:366]: 1 new connections added.
[INFO] [nordnm.nordnm:191]: 'us807 [normal] [tcp]' set as auto-connect server.
[INFO] [nordnm.nordnm:145]: Network kill-switch set successfully!
[INFO] [nordnm.networkmanager:20]: NetworkManager restarted successfully!
```

### When I setup my settings with us and nl
```
[Countries]
# simply write country codes separated by spaces e.g. country-blacklist = gb us
country-blacklist = 

# same as above. if this is non-empty, the blacklist is ignored
country-whitelist = us nl

[Categories]
dedicated-ip-servers = false
p2p = false
double-vpn = false
standard-vpn-servers = true
anti-ddos = false
onion-over-vpn = false

[Protocols]
tcp = true
udp = false

[Benchmarking]
ping-attempts = 5
```
It can't find active servers:
```
$ sudo nordnm -uska nl normal tcp
     _   _               _ _   _ ___  ___
    | \ | |             | | \ | ||  \/  |
    |  \| | ___  _ __ __| |  \| || .  . |
    | . ` |/ _ \| '__/ _` | . ` || |\/| |
    | |\  | (_) | | | (_| | |\  || |  | |
    \_| \_/\___/|_|  \__,_\_| \_/\_|  |_/   v0.0a13 (Latest)

[INFO] [nordnm.nordnm:160]: Downloading latest NordVPN OpenVPN configuration files to '/home/peteruithoven/.nordnm/configs/'.
[INFO] [nordnm.nordnm:307]: Checking for new connections to import...
[WARNING] [nordnm.nordnm:327]: Active VPNs and/or kill-switch disabled for accurate benchmarking. Your connection is not secure until these are re-enabled.
[INFO] [nordnm.nordnm:329]: Benchmarking servers...
[INFO] [nordnm.nordnm:337]: Benchmarking complete. Took 13.76 seconds.
[INFO] [nordnm.nordnm:201]: Auto-Connect file removed.
[INFO] [nordnm.nordnm:209]: Removing all active connections...
[INFO] [nordnm.nordnm:366]: 1 new connections added.
[WARNING] [nordnm.nordnm:194]: No active server found matching (nl, normal, tcp). Check your input and try again.
[INFO] [nordnm.nordnm:145]: Network kill-switch set successfully!
```

While writing this issue and placing some debug statements I figured out that by not enabling P2P I was excluding all nl servers since they have both normal and P2P category? 

The get server list function also filters on valid categories and protocols. Maybe there should be a separate error when the list couldn't be retrieved? 
And a separate error that explains that there where no servers found that just support the enabled countries, categories and protocols. And maybe that servers that support disabled categories will be ignored. I realize this will be a difficult one to write correctly and understandably. 

Maybe `Enable category 'P2P'? (y/n):` doesn't explain well enough that servers that also support this category will be excluded. If I understand the code it works by checking of one of the categories is invalid, so asking whether "server that (also) support category ... should be excluded" might be more appropriate?

**Comment by Chadsr** [comment]
**Valid Server Logic**
I believe my original motive for the logic _"If the server has a category that we don't want, ignore it"_, over _"If the server has a category we want, use it"_ was that, for example, if you want only standard VPN servers, it's best to exclude servers that also have P2P, since the P2P usage may impact the server's performance drastically.

As you have just pointed out, this logic has flaws when there are no servers that are only "standard".

I am now thinking that switching back to the _"If the server has a category we want, use it"_ logic seems like the best option. Especially since server load in the scoring should in theory take into account any performance issues caused by P2P usage, for example.

This will however mean, that even if you disable some categories, such as P2P. P2P servers will still be synchronised if they satisfy an enabled category.

I will put some more thought into this and do some experimentation before pushing any changes.

**Error Message**
Regarding the false error message, I have pushed a change which informs the user that no servers could be found matching their settings, when that is the case.

**Comment by peteruithoven** [comment]
I think the improved errors will help a great deal. 
I understand the predicament... If it's just performance and not any security / privacy consideration I would go for "If the server has a category we want, use it", because it improves usability. But then again, I'm a still a VPN noob.

**Comment by Chadsr** [comment]
Both of these changes (along with many other fixes) are in v0.0a14 (#38) and will be available shortly.

Thanks again for your feedback. It's really appreciated!
